checker.cs

```diff
@@ -1,38 +1,86 @@
 using System;
 using System.Diagnostics;
 
+
 class Checker
+
 {
-    static bool batteryIsOk(float temperature, float soc, float chargeRate) {
-        if(temperature < 0 || temperature > 45) {
+  
+    public static bool temperatureIsOk(float temperature)
+    {
+      if (temperature< 0 || temperature> 45)
+        {
             Console.WriteLine("Temperature is out of range!");
             return false;
-        } else if(soc < 20 || soc > 80) {
+        }
+            return true;
+    }
+
+
+    public static bool socIsOk(float soc)
+    {
+        if (soc < 20 || soc > 80)
+        {
             Console.WriteLine("State of Charge is out of range!");
             return false;
-        } else if(chargeRate > 0.8) {
+        }
+            return true;
+    }
+
+    public static bool chargeRateIsOk(float chargeRate)
+    {
+        if (chargeRate > 0.8)
+        {
             Console.WriteLine("Charge Rate is out of range!");
             return false;
         }
-        return true;
+       
+            return true;
+    }
+
+    static bool batteryIsOk(Func<float, bool> temperatureIsOk, float temperature, Func<float, bool> socIsOk, float soc, Func<float, bool> chargeRateIsOk, float chargeRate)
+    {
+
+        return temperatureIsOk(temperature);
+        return socIsOk(soc);
+        return chargeRateIsOk(chargeRate);
     }
 
-    static void ExpectTrue(bool expression) {
-        if(!expression) {
+    public static void IsExpectTrueOk(bool expression)
+    {
+        if (!expression)
+        {
             Console.WriteLine("Expected true, but got false");
             Environment.Exit(1);
         }
     }
-    static void ExpectFalse(bool expression) {
-        if(expression) {
+        static void ExpectTrue(Action<bool> IsExpectTrueOk, bool expression)
+    {
+        IsExpectTrueOk(expression);
+    }
+    static void IsExpectFalseOk(bool expression)
+    {
+        if (expression)
+        {
             Console.WriteLine("Expected false, but got true");
             Environment.Exit(1);
         }
     }
-    static int Main() {
-        ExpectTrue(batteryIsOk(25, 70, 0.7f));
-        ExpectFalse(batteryIsOk(50, 85, 0.0f));
+
+    static void ExpectFalse(Action<bool> IsExpectFalseOk, bool expression)
+    {
+        IsExpectFalseOk(expression);
+    }
+
+
+
+
+    static int Main()
+    {
+        ExpectTrue(IsExpectTrueOk, batteryIsOk(temperatureIsOk,25,socIsOk, 70, chargeRateIsOk,0.7f));
```

Comment:
Test cases are not sufficient to get full coverage. Also these test cases need to strengthened with variety of input values.